lib/parse.js

```diff
@@ -1,5 +1,77 @@
 const util = require('./util')
 
+const ParseState = {
+    start: 0,
+    beforePropertyName: 1,
+    afterPropertyName: 2,
+    beforePropertyValue: 3,
+    afterPropertyValue: 4,
+    beforeArrayValue: 5,
+    afterArrayValue: 6,
+    end: 7,
+}
+
+const LexState = {
+    default: 8,
+    comment: 9,
+    multiLineComment: 10,
+    multiLineCommentAsterisk: 11,
+    singleLineComment: 12,
+    value: 13,
+    identifierNameStartEscape: 14,
+    identifierName: 15,
+    identifierNameEscape: 16,
+    sign: 17,
+    zero: 18,
+    decimalInteger: 19,
+    decimalPointLeading: 20,
+    decimalPoint: 21,
+    decimalFraction: 22,
+    decimalExponent: 23,
+    decimalExponentSign: 24,
+    decimalExponentInteger: 25,
+    hexadecimal: 26,
+    hexadecimalInteger: 27,
+    string: 28,
+    start: ParseState.start,
+    beforePropertyName: ParseState.beforePropertyName,
+    afterPropertyName: ParseState.afterPropertyName,
+    beforePropertyValue: ParseState.beforePropertyValue,
+    afterPropertyValue: ParseState.afterPropertyValue,
+    beforeArrayValue: ParseState.beforeArrayValue,
+    afterArrayValue: ParseState.afterArrayValue,
+    end: ParseState.end,
+}
+
+const Token = {
+    eof: 0,
+    punctuator: 1,
+    null: 2,
+    boolean: 3,
+    numeric: 4,
+    identifier: 5,
+    numeric: 6,
+    string: 7,
+    punctuator: 8,
+}
+
+const tokenRef = {
+    type: -1,
+    value: '',
+    line: -1,
+    column: -1,
+}
+
+const FIRST_DOUBLE_LENGTH_CODE = 2 ** 16
```

Comment:
json5@2 still supports Node.js v6, which doesn't support the `**` operator. Please hardcode the value for now. Otherwise, this PR will have to wait until json5@3.